lib/spack/spack/directory_layout.py

```diff
@@ -264,13 +264,18 @@ def create_install_directory(self, spec):
         from spack.package_prefs import get_package_group
         group = get_package_group(spec)
         perms = get_package_dir_permissions(spec)
-        mkdirp(spec.prefix, mode=perms)
+        perms_intermediate = 0o777
+
+        mkdirp(spec.prefix, mode=perms, mode_intermediate=perms_intermediate)
         if group:
             chgrp(spec.prefix, group)
             # Need to reset the sticky group bit after chgrp
             os.chmod(spec.prefix, perms)
 
-        mkdirp(self.metadata_path(spec), mode=perms)
+        mkdirp(self.metadata_path(spec),
+               mode=perms,
+               mode_intermediate=perms_intermediate)
```

Comment:
I believe this is unnecessary, as the metadata path is only one directory below the prefix. Maybe to future proof it we should set the intermediate directories, but if so it should be to the same permissions as the prefix because it's within the prefix.

Reply:
I guess this change is less necessary, since we can rely on the prefix as gatekeeper. But that seems non-ideal.

Reply:
Ok, for me it was just to be sure that all package related things worked with right permissions, without assuming that metadata path has to be stored inside the package folder.

But if we can assume that, I can remove it and decide to add a comment for future reference.

Reply:
Ok, for me it was just to be sure that all package related things worked with right permissions, without assuming that metadata path has to be stored inside the package folder.

But if we can assume that, I can remove it and decide to add a comment for future reference.

Reply:
I think it's safe to leave this as-is for now.